### PR TITLE
Swapped reach router with next router

### PR DESCRIPTION
**Pull request by Valioesi**
I removed the reach router dependency and the only appearance (for now). We will probably have to change more imports as soon as we drag in the rest of the components.

### CODE REVIEW

File: components/Login/RequestLoginCode/index.tsx [hunk 1]
```diff
@@ -102,7 +103,7 @@ export const RequestLoginCodeWithEmail = ({
               // a the current url to the browser history, so we go back
               // to email form after user presses back button in login code form.
               // If the hash was already #code we want to replace the current page in the stack.
-              navigate('#code', { replace: window.location.hash === '#code' });
```
Comment (nmauersberg) [inline components/Login/RequestLoginCode/index.tsx:105]: @Valioesi Login still works as expected, do you think the check is still relevant?

**Review by nmauersberg — APPROVED** [review]
I have found three more occurrences of the @reach/router and removed them. And I rebased with main, because of merge conflicts in yarn.lock. I hope it's okay, that I just committed that, let me know, if you'd rather have requested changes in such a case! :)

I'd be okay with the pr like this, but I am not sure, if the "#code" part covers all cases. Left a comment on that line.

File: components/Login/RequestLoginCode/index.tsx (hunk 1 above)
Comment (Valioesi) [reply components/Login/RequestLoginCode/index.tsx:105]: Thanks! I also just stumbled upn those occurences. Weird, that the global search did not find them...
I am fine with you just making the changes :)

Comment (Valioesi) [reply components/Login/RequestLoginCode/index.tsx:105]: We don't want to use the replace function everytime though :) 
I'll make some changes.

**Comment by nmauersberg** [comment]
@Valioesi Cool, looks good! I rebased it with main, to remove the conflicts in the yarn.lock file due to a circles bump.
Let's merge! (Or rebase, hehe :)

**Pull request merged**